src/Microsoft.Diagnostics.Runtime/DataTarget.cs

```diff
@@ -195,15 +195,40 @@ private ImmutableArray<ClrInfo> GetOrCreateClrVersions()
 
             if (_clrs.IsDefault)
             {
+                IEnumerable<ModuleInfo> modules = EnumerateModules();
+                IEnumerable<ClrInfo>? clrs = null;
+
+                // First try the SpecialDiagInfo block short cut to find a supported runtime
+                if (!_target.ForceCompleteRuntimeEnumeration && DataReader.TargetPlatform != OSPlatform.Windows)
+                {
+                    if (SpecialDiagInfo.TryReadSpecialDiagInfo(DataReader, out SpecialDiagInfo info) && info.RuntimeBaseAddress != 0)
+                    {
+                        Debug.Assert(_modules != null);
+                        
+                        // The above EnumerateModules() call populated (the sorted) _modules array
+                        int found = _modules!.Search(info.RuntimeBaseAddress, (m, k) => m.ImageBase.CompareTo(k));
+                        if (found > 0)
+                        {
+                            // Run the module through the clr info providers to see if it is an app model/runtime supported by CLRMD
+                            clrs = from clrInfo in s_clrInfoProviders.Select(provider => provider.ProvideClrInfoForModule(this, _modules![found])).Where(clrInfo => clrInfo != null)
```

Comment:
This sets the `clrs` variable to be non-null, skipping the below.  Aside from the question I have below, note that s_clrInfoProviders could all return null.  Meaning we will have a non null `clrs` with an empty enumeration, so it will skip the full enumeration below, even though this method failed.

I guess this couldn't happen in practice, but maybe it would be better to change 'clrs' to a List<> and build it to check what we have instead of leaving `clrs` as an enumeration in this new logic.

Reply:
Yes, it should for an empty list instead of using null.  I'll push a change.